### PR TITLE
Show build initiator on Workspace Build page

### LINKED ISSUE #2029: Show Initiator in Build Logs
There have been cases where either:
1. Auto-start/Auto-stop triggers a start/stop
2. An admin user triggers a start/stop

And it's unclear why a workspace transitioned state. I used this opaqueness to troll Kyle for multiple days.

I suggest we add an ~"Actor"~ Initiated By column to the Timeline table in the Workspace page and an Actor field on the Build Log page.

## Implementation notes: 

We should add a system user here to report in the logs for what user is taking what actions. 

## AC
Show a human readable name for the initiator ID for all builds. 
Add another column in the build logs for initiator
Testing: storybook for UI, human readable name, check that the name appears on the page, unit test for API returning human readable name

### LINKED ISSUE #2410: Show the reason for workspace action on build page
Discussion on #2029 with @ketang to show the reason for a workspace action:

We might have multiple reasons to perform a particular action. For instance, a workspace could get (re)started or built as part of administrators forcing all workspaces to update the project.

This needs more discussion:
- Do we ask the user to input the reason when they are taking an action?
- Should we intelligently figure out ourselves what the reason would be? Based on who did it and what they did.
- What should the UX for this be?

### PR DESCRIPTION
**Pull request by AbhineetJain**
This PR builds on #2438 and shows the workspace build initiator on the Workspace Build page.

## Subtasks

- [x] add a column in the stats to show build initiator
- [x] format the initiator based on reason
- [x] add stories

Fixes #2029 
Fixes #2410 

## Screenshots

Screen Shot 2022-06-16 at 7 27 50 PM

Screen Shot 2022-06-16 at 7 20 28 PM

Screen Shot 2022-06-16 at 7 20 49 PM

### CODE REVIEW

File: site/src/util/workspace.ts [hunk 1]
```diff
@@ -186,6 +186,37 @@ export const getDisplayWorkspaceBuildStatus = (
...
+    case "autostart":
+      return {
+        color: theme.palette.secondary.dark,
+        initiatedBy: DisplayWorkspaceBuildInitiatedByLanguage.autostart,
+      }
+    case "autostop":
+      return {
+        color: theme.palette.secondary.dark,
+        initiatedBy: DisplayWorkspaceBuildInitiatedByLanguage.autostop,
+      }
```
Comment (Kira-Pilot) [inline site/src/util/workspace.ts:216]: My feeling (and I know this is boring of me) is that we probably don't need to highlight these values in purple. I am concerned we will be drawing too much attention to information that, while important, is not particularly actionable. That being said, I love that you tried a different design! Maybe post this in the dev-ux channel to get more opinions - I'm not a designer by any stretch of the imagination haha.

**Review by Kira-Pilot — APPROVED** [review]

**Comment by AbhineetJain** [comment]
@Kira-Pilot Thanks for the suggestions. I am merging this for now. Let's create another issue to discuss the design, and if we reach a consensus, we can update it.

**Comment by ketang** [comment]
@AbhineetJain I'm not seeing this on dev.coder.com.  I'm looking at both the workspaces (plural) and workspace (singular) pages.

**Comment by AbhineetJain** [comment]
@ketang The initiator is shown for each build of the workspace. You may check any of the workspace builds from the timeline table.

**Comment by ketang** [comment]
Aaah okay I see now.  The description says "Workspace Build" page, but I was looking at the Workspace page.  I'll file an issue for that.

**Comment by ketang** [comment]
#2865 2865

**Pull request merged**